gprofiler/profilers/profiler_base.py

```diff
@@ -171,6 +171,12 @@ def _profiling_error_stack(
     def snapshot(self) -> ProcessToProfileData:
         processes_to_profile = self._select_processes_to_profile()
         logger.debug(f"{self.__class__.__name__}: selected {len(processes_to_profile)} processes to profile")
+        if self._profiler_state.processes_to_profile is not None:
+            processes_to_profile = [
+                process for process in processes_to_profile if process in self._profiler_state.processes_to_profile
+            ]
+            if len(processes_to_profile) > 0:
```

Comment:
This `if` should be before filtering. What I suggested is, if we have 0 selected processes to begin with, then obviously we're left with 0 after filtering. But if we have 2 selected processes and both are filtered out, then the current code won't print that 0 are left.

The condition can be `if self._profiler_state.processes_to_profile is not None and len(processes_to_profile) > 0:`.